hl/test/CMakeLists.txt

```diff
@@ -87,16 +87,25 @@ endif ()
 # This executable is used to generate test files for the test_ds test.
 # It should only be run during development when new test files are needed
 # --------------------------------------------------------------------
-if (HDF5_BUILD_GENERATORS AND BUILD_STATIC_LIBS)
+if (HDF5_BUILD_GENERATORS)
   add_executable (hl_gen_test_ds gen_test_ds.c)
   target_compile_options(hl_gen_test_ds PRIVATE "${HDF5_CMAKE_C_FLAGS}")
   target_include_directories (hl_gen_test_ds PRIVATE "${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  TARGET_C_PROPERTIES (hl_gen_test_ds STATIC)
-  target_link_libraries (hl_gen_test_ds PRIVATE
-      ${HDF5_HL_LIB_TARGET}
-      ${HDF5_TEST_LIB_TARGET}
-      ${HDF5_LIB_TARGET}
-  )
+  if (NOT BUILD_SHARED_LIBS)
```

Comment:
Slightly different logic for static vs. shared here by using NOT, but this just matches the existing logic in the file.